utils/concurrentmap/concurrent_map.go

```diff
@@ -1,44 +1,44 @@
-package unittest
+package concurrentmap
 
 import "sync"
 
-// ProtectedMap is a thread-safe map.
-type ProtectedMap[K comparable, V any] struct {
+// ConcurrentMap is a thread-safe map.
+type ConcurrentMap[K comparable, V any] struct {
 	mu sync.RWMutex
 	m  map[K]V
 }
 
-// NewProtectedMap returns a new ProtectedMap with the given types
-func NewProtectedMap[K comparable, V any]() *ProtectedMap[K, V] {
-	return &ProtectedMap[K, V]{
+// NewConcurrentMap returns a new ConcurrentMap with the given types
+func NewConcurrentMap[K comparable, V any]() *ConcurrentMap[K, V] {
```

Comment:
Now that this is in its own module, you don't need to qualify the new method
```suggestion
func New[K comparable, V any]() *ConcurrentMap[K, V] {
```